src/cli/self_update/windows.rs

```diff
@@ -144,7 +144,7 @@ pub fn do_add_to_path() -> Result<()> {
     _apply_new_path(new_path)
 }
 
-fn _apply_new_path(new_path: Option<String>) -> Result<()> {
+fn _apply_new_path(new_path: Option<Vec<u16>>) -> Result<()> {
```

Comment:
This should be an [OsString](https://doc.rust-lang.org/std/ffi/struct.OsString.html) as thats what we can accurately get from the Path type, and we'll need to fix up the home crate, and thats the type I'll ask for there as well to support this there.